databricks/koalas/tests/indexes/test_datetime.py

```diff
@@ -128,3 +129,38 @@ def test_strftime(self):
             self.assert_eq(
                 kidx.strftime(date_format="%B %d, %Y"), pidx.strftime(date_format="%B %d, %Y")
             )
+
+    def test_indexer_between_time(self):
+        with ks.option_context("compute.shortcut_limit", 0):
```

Comment:
Could you try both `compute.shortcut_limit` is the default and `0`?

Maybe we can:

```py
def test_indexer_between_time(self):
    for kidx, pidx in self.idx_pairs:
...

def test_indexer_between_time_without_shortcut(self):
    with ks.option_context("compute.shortcut_limit", 0):
         self.test_indexer_between_time()
```

Reply:
Or explicitly add a type hint to the UDF for `apply_batch` to avoid falling into shortcut path?

```py
def pandas_between_time(pdf) -> ks.DataFrame[int]:
```

Reply:
Adding type hints would save us from inferring schema, but lose the chance of shortcuts.

Do we always add type hints to UDF defined in the code base?

Reply:
Now that we added the return type hint, we can remove `with ks.option_context("compute.shortcut_limit", 0):`.

Reply:
> Do we always add type hints to UDF defined in the code base?

Yes, it'd be preferred in general.

Reply:
Sounds good!